src/Illuminate/Support/Str.php

```diff
@@ -590,6 +590,33 @@ public static function orderedUuid()
         return $factory->uuid4();
     }
 
+    /**
+     * Check if $className is a valid PHP classname.
+     *
+     * @see https://www.php.net/manual/en/language.oop5.basic.php
+     *
+     * @param string $className
+     * @return bool
+     */
+    public static function validClassName($className)
+    {
+        // This may be a FQCN, so we split it into parts to check each
+        $parts = explode('\\', ltrim($className, '\\'));
```

Comment:
the `ltrim($className, '\\')` should probably only remove the first backslash.

i.e. Str::validClassName("\\\\\\\\\test");

Reply:
That is a very good point - thanks :-)